palette/moreland/smooth_test.go

```diff
@@ -173,25 +163,60 @@ func fracToByte(v float64) uint8 {
 func testRGB(t *testing.T, i int, label string, c1 color.Color, c2 [3]float64) {
 	// The test tolerance is the precision of a uint8 expressed as a uint32.
 	const tolerance = 1.0 / 256.0 * 65535.0
-
-	r, g, b, a := c1.RGBA()
 	c3 := color.NRGBA{
 		R: fracToByte(c2[0]),
 		G: fracToByte(c2[1]),
 		B: fracToByte(c2[2]),
 		A: 255,
 	}
-	wr, wg, wb, wa := c3.RGBA()
-	if math.Abs(float64(r)-float64(wr)) > tolerance {
-		t.Errorf("%d %s R: want %d but have %d", i, label, wr, r)
+	if !similar(c1, c3, tolerance) {
+		t.Errorf("%d %s: want %+v but have %+v", i, label, c1, c3)
+	}
+}
+
+func TestSmoothDiverging_At(t *testing.T) {
+	// The test tolerance is the precision of a uint8 expressed as a uint32.
+	const tolerance = 1.0 / 256.0 * 65535.0
```

Comment:
I don't understand this at all.

Ah, I think I get it now. You are asking for a single bit of tolerance. Can you do this at a package level and provide a more explanatory comment.